runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -246,20 +249,24 @@ public void invalidateClientCache() {
     }
 
     /**
-     * Fetch an address for insertion into the cache.
+     * Fetch a collection of addresses for insertion into the cache.
      *
-     * @param addresses An address to read from.
-     * @return A result to be cached. If the readresult is empty,
-     * This entry will be scheduled to self invalidate.
+     * @param addresses collection of addresses to read from.
+     * @return A result to be cached
      */
     private @Nonnull Map<Long, ILogData> cacheFetch(Iterable<Long> addresses) {
-        return StreamSupport.stream(addresses.spliterator(), true)
-                .map(address -> layoutHelper(l ->
-                        l.getReplicationMode(address).getReplicationProtocol(runtime)
-                        .read(l, address)))
-                .collect(Collectors.toMap(ILogData::getGlobalAddress, r -> r));
-    }
+        //turn the addresses into Set for now to satisfy signature requirement down the line
```

Comment:
I didn't want to make any changes to method signatures at this point, since I wasn't sure if there'd be any changes required when I implement random reads.